rust/agama-cli/src/logs.rs

```diff
@@ -0,0 +1,257 @@
+        let file_path = self.to();
+        let output = Command::new(cmd_parts[0])
+            .args(cmd_parts[1..].iter())
+            .output()
+            .expect("Failed to run the command");
+        let mut file_stdout = File::create(format!("{}.out.log", file_path.display()))?;
+        let mut file_stderr = File::create(format!("{}.err.log", file_path.display()))?;
+
+        file_stdout.write_all(&output.stdout)?;
+        file_stderr.write_all(&output.stderr)?;
+
+        Ok(())
+    }
+}
+
+fn store(verbose: bool) -> Result<(), io::Error> {
+    if !Uid::effective().is_root() {
+        panic!("No Root, no logs. Sorry.");
+    }
+
+    // 0. preparation, e.g. in later features some log commands can be added / excluded per users request or
+    let commands = DEFAULT_COMMANDS;
+    let paths = DEFAULT_PATHS;
+    let result = format!("{}.{}", DEFAULT_RESULT, DEFAULT_COMPRESSION.1);
+    let noisy = verbose;
+    let compression = DEFAULT_COMPRESSION.0;
+    let mut log_sources: Vec<Box<dyn LogItem>> = Vec::new();
+
+    showln(noisy, "Collecting Agama logs:");
+
+    // 1. create temporary directory where to collect all files (similar to what old save_y2logs
+    // does)
+    let tmp_dir = TempDir::new(DEFAULT_TMP_DIR)?;
+
+    // 2. collect existing / requested paths which should already exist
+    showln(noisy, "\t- proceeding well known paths");
+    for path in paths {
+        // assumption: path is full path
+        if Path::new(path).try_exists().is_ok() {
+            log_sources.push(Box::new(LogPath::new(path, tmp_dir.path())));
+        }
+    }
+
+    // 3. some info can be collected via particular commands only
+    showln(noisy, "\t- proceeding output of commands");
+    for cmd in commands {
+        log_sources.push(Box::new(LogCmd::new(cmd, tmp_dir.path())));
+    }
+
+    // 4. store it
+    showln(true, format!("Storing result in: \"{}\"", result).as_str());
+
+    for log in log_sources.iter() {
+        show(
+            noisy,
+            format!("\t- storing: \"{}\" ... ", log.from()).as_str(),
+        );
+
+        // for now keep directory structure close to the original
+        // e.g. what was in /etc will be in /<tmp dir>/etc/
+        let res = match fs::create_dir_all(log.to().parent().unwrap()) {
+            Ok(_p) => match log.store() {
+                Ok(_p) => "[Ok]",
+                Err(_e) => "[Failed]",
+            },
+            Err(_e) => "[Failed]",
+        };
+
+        showln(noisy, format!("{}", res).as_str());
+    }
+
+    let compress_cmd = format!(
+        "tar -c -f {} --warning=no-file-changed --{} --dereference -C {} .",
+        result,
+        compression,
+        tmp_dir.path().display()
+    );
+    let cmd_parts = compress_cmd.split_whitespace().collect::<Vec<&str>>();
+
+    Command::new(cmd_parts[0])
+        .args(cmd_parts[1..].iter())
+        .status()
+        .expect("failed creating the archive");
```

Comment:
Panicking :-(

Reply:
consequence of learning ;-) will modify, I see into it more now ;-)